astroquery/esasky/core.py

```diff
@@ -0,0 +1,624 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import json
+import os
+import tempfile
+import tarfile
+import sys
+
+from astropy.extern import six
+import astropy.io.votable as votable
+from astropy.table import Table
+from astropy.io import fits
+import astropy.units
+
+from ..query import BaseQuery
+from ..utils import commons
+from ..utils import async_to_sync
+from . import conf
+from ..exceptions import TableParseError
+
+@async_to_sync
+class ESASkyClass(BaseQuery):
+
+    URLbase = conf.urlBase
+    TIMEOUT = conf.timeout
+    
+    __FITS_STRING = ".fits"
+    __FTZ_STRING = ".FTZ"
+    __TAR_STRING = ".tar"
+    __ALL_STRING = "all"
+    __CATALOGS_STRING = "catalogs"
+    __OBSERVATIONS_STRING = "observations"
+    __MISSION_STRING = "mission"
+    __TAP_TABLE_STRING = "tapTable"
+    __TAP_NAME_STRING = "tapName"
+    __FILTER_STRING = "filter"
+    __LABEL_STRING = "label"
+    __OBSERVATION_ID_STRING = "observation_id"
+    __METADATA_STRING = "metadata"
+    __MAPS_STRING = "Maps"
+    __PRODUCT_URL_STRING = "product_url"
+    __SOURCE_LIMIT_STRING = "sourceLimit"
+    __POLYGON_NAME_STRING = "polygonNameTapColumn"
+    __POLYGON_RA_STRING = "polygonRaTapColumn"
+    __POLYGON_DEC_STRING = "polygonDecTapColumn"
+    __POS_TAP_STRING = "posTapColumn"
+    __ORDER_BY_STRING = "orderBy"
+    __IS_SURVEY_MISSION_STRING = "isSurveyMission"
+    __ZERO_ARCMIN_STRING = "0 arcmin"
+
+    __HERSCHEL_STRING = 'herschel'
+    __HST_STRING = 'hst'
+    __INTEGRAL_STRING = 'integral'
+
+    
+    def list_maps(self):
+        """
+        Get a list of the mission names of the available observations in ESASky
+        """
+        return self._json_object_field_to_list(self._get_observation_json(cache = True), self.__MISSION_STRING)
+    
+    def list_catalogs(self):
+        """
+        Get a list of the mission names of the available catalogs in ESASky
+        """
+        return self._json_object_field_to_list(self._get_catalogs_json(cache = True), self.__MISSION_STRING)    
+    
+    def query_object_maps(self, position, missions = __ALL_STRING, get_query_payload = False, cache = True):
```

Comment:
Minor style comments:

- in function definitions, we prefer not to have spaces surrounding `=` signs.

- when possible, try to wrap lines at <= 80 characters.